components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -530,6 +530,212 @@ func (backend *ES2Backend) GetReport(esIndex string, reportId string,
+		elastic.NewTermsAggregation().Field("profiles.name"))
+
+	profileInfoAgg.SubAggregation("sha",
+		elastic.NewTermsAggregation().Field("profiles.sha256"))
+
+	profileInfoAgg.SubAggregation("title",
+		elastic.NewTermsAggregation().Field("profiles.title"))
+
+	profileInfoAgg.SubAggregation("version",
+		elastic.NewTermsAggregation().Field("profiles.version"))
+
+	controlTermsAgg.SubAggregation("profile", profileInfoAgg)
+
+	//todo handle passed in regex too
+	filteredControls := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().Must(elastic.NewRegexpQuery("profiles.controls.id", ".*")))
+
+	filteredControls.SubAggregation("control", controlTermsAgg)
+
+	controlsAgg := elastic.NewNestedAggregation().Path("profiles.controls")
+	controlsAgg.SubAggregation("filtered_controls", filteredControls)
+
+	searchSource.Aggregation("controls", controlsAgg)
+
+	source, err := searchSource.Source()
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("%s unable to get Source", myName))
+	}
+	LogQueryPartMin(esIndex, source, fmt.Sprintf("%s query", myName))
+
+	searchResult, err := client.Search().
+		SearchSource(searchSource).
+		Index(esIndex).
+		Do(context.Background())
+
+	if err != nil {
+		logrus.Error("getProfileSuggestions search failed")
+		return nil, err
+	}
+
+	logrus.Debugf("Search query took %d milliseconds\n", searchResult.TookInMillis)
+
+	if outerControlsAggResult, found := searchResult.Aggregations.Nested("controls"); found {
+		if filteredControls, found := outerControlsAggResult.Aggregations.Filter("filtered_controls"); found {
+			if controlBuckets, found := filteredControls.Aggregations.Terms("control"); found && len(controlBuckets.Buckets) > 0 {
+				for _, controlBucket := range controlBuckets.Buckets {
+					contListItem := reportingapi.ControlItem{}
+					contListItem.Id = controlBucket.Key.(string)
+					controlSummary := &reportingapi.ControlSummary{
+						Passed:  &reportingapi.Total{},
+						Skipped: &reportingapi.Total{},
+						Failed:  &reportingapi.Failed{},
+					}
+
+					profileMin := &reportingapi.ProfileMin{}
+
+					if aggResult, found := controlBucket.Aggregations.Terms("title"); found {
+						//there can only be one
+						bucket := aggResult.Buckets[0]
+						title, ok := bucket.Key.(string)
+						if !ok {
+							logrus.Errorf("could not convert the value of title: %v, to a string!", bucket)
+						}
+						contListItem.Title = title
+					}
+
+					if impactAggResult, found := controlBucket.Aggregations.Terms("impact"); found {
+						//there can only be one
+						impactBucket := impactAggResult.Buckets[0]
+						impactAsNumber, ok := impactBucket.Key.(float64)
+						if !ok {
+							logrus.Errorf("could not convert the value of impact: %v, to a float!", impactBucket)
+						}
+						contListItem.Impact = float32(impactAsNumber)
+						controlSummary.Total = int32(impactBucket.DocCount)
+					}
+
+					if profileResult, found := controlBucket.Aggregations.ReverseNested("profile"); found {
```

Comment:
[minor] I'm wondering if there is a way to make some of this inner loop more clear.  One small improvement might be to move the declaration of `profileMin` right before this loop that uses it, and then assign it to contListItem.Profile right after this section rather than below at 705.

Reply:
done.. thanks!